src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -51,47 +49,40 @@ class ResConfig:
     def __init__(
         self,
         user_config_file: Optional[str] = None,
-        config: Optional[Dict[ConfigKeys, Any]] = None,
         config_dict: Optional[Dict[ConfigKeys, Any]] = None,
     ):
 
-        self._assert_input(user_config_file, config, config_dict)
+        self._assert_input(user_config_file, config_dict)
         self.user_config_file = user_config_file
 
-        self._errors, self._failed_keys = None, None
+        self._errors = None
         self._templates = []
-        if user_config_file or config:
+        if user_config_file:
             self._alloc_from_content(
                 user_config_file=user_config_file,
-                config=config,
             )
         else:
             self._alloc_from_dict(config_dict=config_dict)
 
         self._validate()
 
-    def _assert_input(self, user_config_file, config, config_dict):
-        configs = sum(
-            1 for x in [user_config_file, config, config_dict] if x is not None
-        )
-
-        if configs > 1:
+    def _assert_input(self, user_config_file, config_dict):
+        user_config_file_given = user_config_file is not None
+        config_dict_given = config_dict is not None
+        if user_config_file_given and config_dict_given:
             raise ValueError(
                 "Attempting to create ResConfig object with multiple config objects"
             )
 
-        if configs == 0:
+        if not user_config_file_given and not config_dict:
```

Comment:
Was the intent to test `config_dict_given` instead of `config_dict` ? The end result is still the same though.